pageserver/src/virtual_file.rs

```diff
@@ -240,8 +206,9 @@ impl VirtualFile {
             timelineid = "*".to_string();
         }
         let (handle, mut slot_guard) = get_open_files().find_victim_slot();
-        let file = STORAGE_IO_TIME
-            .with_label_values(&["open", &tenantid, &timelineid])
+        let metrics = Arc::new(TimelineMetrics::new(&tenantid, &timelineid));
+        let file = metrics
+            .storage_io_time("open")
```

Comment:
Seems pointless to create the new TimelineMetrics object here. It's only used to call `storage_io_time`, and `storage_io_time` just fetches the tenant and timeline ID from it and performs a lookup in STORAGE_IO_TIME anyway.

Actually I think this works incorrectly for metrics cleanup. If the VirtualFile is dropped, its 'metrics' will be dropped, and it will delete the metrics. Even if the Timeline object is still alive, with its own TimelineMetrics instance.